sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkReadWriteLogRecord.java

```diff
@@ -25,7 +26,7 @@ class SdkReadWriteLogRecord implements ReadWriteLogRecord {
   private final Resource resource;
   private final InstrumentationScopeInfo instrumentationScopeInfo;
   private final long epochNanos;
-  private final SpanContext spanContext;
+  private final Context context;
```

Comment:
I think(?) the reason that SdkSpan only contains `SpanContext` instead of the full `Context` was to limit memory retention, @jkwatson do you recall anything?

Reply:
Ok so there is a subtle difference between holding it in the `SdkReadWriteLogRecord` and holding it in the `SdkLogRecordBuilder` such that it can be passed to each log processor as an argument `LogRecordProcessor#onEmit`.

If its held in `SdkReadWriteLogRecord`, there's a reference to `Context` until `BatchLogRecordProcessor` flushes and converts each log record to `LogRecordData`, so the reference sticks around for essentially the shorter of the batch interval or the time it takes to fill the batch queue.

If instead the context is held in `SdkLogRecordBuilder` and passed as an argument to each `LogRecordProcessor#onEmit()`, then the reference only has to stick around until the last processor has synchronously returns from `onEmit`. This allows the context to be garbage collected while the corresponding log record is still sitting in the `BatchLogRecordProcessor` queue.

I think this is a good enough reason to include as an argument to `onEmit()` instead of having a getter on `SdkReadWriteLogRecord`.